src/os/bluestore/BlueStore.cc

```diff
@@ -16130,6 +16161,226 @@ void BlueStore::BlueStoreThrottle::complete(TransContext &txc)
 }
 #endif
 
+BlueStore::BlueStoreSlowFastCoDel::BlueStoreSlowFastCoDel(CephContext *_cct) :
+  fast_timer(_cct, fast_timer_lock), slow_timer(_cct, slow_timer_lock) {}
+
+BlueStore::BlueStoreSlowFastCoDel::~BlueStoreSlowFastCoDel() {
+  std::lock_guard l1{fast_timer_lock};
+  fast_timer.cancel_all_events();
+  fast_timer.shutdown();
```

Comment:
These should probably be scoped so that fast_timer_lock is released prior to taking slow_timer_lock?